Hudl.Mjolnir/Bulkhead/BulkheadFactory.cs

```diff
@@ -45,7 +46,7 @@ public BulkheadFactory(IMetricEvents metricEvents, IBulkheadConfig bulkheadConfi
                         if (_bulkheads.TryGetValue(key, out Lazy<SemaphoreBulkheadHolder> holder) && holder.IsValueCreated)
                         {
                             var bulkhead = holder.Value.Bulkhead;
-                            _metricEvents.BulkheadGauge(bulkhead.Name, "semaphore", _bulkheadConfig.GetMaxConcurrent(key), bulkhead.CountAvailable);
+                            _metricEvents.BulkheadGauge(bulkhead.Name, "semaphore", _config.GetBulkheadConfiguration(key.Name).MaxConcurrent, bulkhead.CountAvailable);
```

Comment:
Would it be better to pass in a `GroupKey` to `GetBulkheadConfiguration`?

Reply:
I had considered that but decided that it'll be easier to implement Configuration in any adapter if that will be just the string. So for simplicity, I've left it as a string.

Reply:
More of an opinion, but I think it's clearer to the calling code if it just has to pass in a `GroupKey`, since then callers don't have to worry about translating a `GroupKey` to its appropriate configuration string - the configuration code can know what to do with the `GroupKey` and do it consistently for any call. `GroupKey` is the abstraction over bulkhead/breaker identifiers and I've always preferred to use that within Mjolnir if it's available instead of raw strings.

Reply:
Not a strong argument, but sort of relevant: http://wiki.c2.com/?StringlyTyped